apps/server/src/modules/tool/external-tool/service/datasheet-pdf.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@nestjs/common';
+import { createPdf, TCreatedPdf } from 'pdfmake/build/pdfmake';
+import pdfFonts from 'pdfmake/build/vfs_fonts';
+import { StyleDictionary } from 'pdfmake/interfaces';
+import { ExternalToolDatasheetTemplateData, ExternalToolParameterDatasheetTemplateData } from '../domain';
+
+@Injectable()
+export class DatasheetPdfService {
+	public generatePdf(templateData: ExternalToolDatasheetTemplateData): Promise<Buffer> {
+		const deactivated: string = templateData.isDeactivated ? 'Dieses Tool ist deaktivert.' : '';
+		const restricted: string =
+			templateData.restrictToContexts && templateData.restrictToContexts.length > 0
+				? `Dieses Tool ist auf folgende Kontexte beschränkt:${templateData.restrictToContexts}`
+				: '';
+
+		const toolTypeParams = [''];
+		if (templateData.toolType === 'OAuth 2.0') {
+			const oauthParam: string = templateData.skipConsent ? `Zustimmung überspringen: ${templateData.skipConsent}` : '';
+			toolTypeParams.push(oauthParam);
+		}
+		if (templateData.toolType === 'LTI 1.1' && templateData.messageType && templateData.privacy) {
+			const ltiParam1 = `Message Type: ${templateData.messageType}`;
+			const ltiParam2 = `Privatsphäre: ${templateData.privacy}`;
+			toolTypeParams.push(ltiParam1, ltiParam2);
+		}
+
+		return new Promise<Buffer>((resolve, reject) => {
+			try {
+				const documentDefinition = {
+					content: [
+						{ text: `Erstellt am ${templateData.createdAt} von ${templateData.creatorName}`, style: 'right-aligned' },
+						'\n',
+						{ text: templateData.instance, style: 'right-aligned' },
+						'\n',
+						{ text: 'Datenblatt', style: ['center-aligned', 'h1'] },
+						{ text: templateData.toolName, style: ['center-aligned', 'h2'] },
+						'\n',
+						{ text: templateData.toolUrl, style: ['center-aligned', 'link'], link: templateData.toolUrl },
+						'\n\n',
+						deactivated,
+						restricted,
+						'\n',
+						`Typ des Tools: ${templateData.toolType}`,
+						toolTypeParams,
+
+						...(templateData?.parameters?.length
```

Comment:
it would probably be more readable when you do this with array.push() in a builder-like pattern.
```
if(templateData.isDeactivated) {
    content.push('Dieses Tool ist deaktivert.')
}

if(templateData.parameters?.length) {
    content.push([
        { text: 'An den Dienst übermittelte Parameter', style: 'h4' },
        '\n',
        ....
    ])
}
...
```